navigator/backend/src/main/scala/com/digitalasset/navigator/store/platform/PlatformStore.scala

```diff
@@ -159,21 +159,34 @@ class PlatformStore(
   }
 
   def connected(state: StateConnected): Receive = {
-    case UpdateParties =>
+    case UpdatePartiesAndUsers =>
       state.ledgerClient.partyManagementClient
         .listKnownParties()
         .map(UpdatedParties(_))
         .pipeTo(self)
+
+      state.ledgerClient.userManagementClient
+        .listUsers() // TODO what token should we pass?
+        .map(UpdatedUsers(_))
+        .pipeTo(self)
       ()
+
+    case UpdatedUsers(users) =>
+      // TODO: should we be able to log in as a user without a primary party?
+      val usersWithPrimaryParties = users.flatMap { user => user.primaryParty.map(p => (user.id, p)) }
+
+      usersWithPrimaryParties.foreach { case (userId, party) =>
+        self ! Subscribe(
+          userId, // TODO: state.parties.keys are a mixed bag: a Party's displayName, a party's name, or a User's id -- is it ok to mix these?
```

Comment:
It’s been a while but iirc they don’t have semantic meaning. They are only used as a human-readable name. So it should be fine but very much in favor of throwing some types at this to differentiate clearly between the 3 (I think there is actually a fourth option: the human readable name specified in the navigator config).